### PR TITLE
Select Field Input Menu scrollable and add Select Field in Filter and Sort

### LINKED ISSUE #3356: Select component should have a scroll and be searchable
I created a custom column in the data model with lots of options.

Settings:
Screenshot 2024-01-10 at 16 15 00

Here's how it was displayed in the UI — I had to scroll the page to scroll the list, there was no "sub scroll":
Screenshot 2024-01-10 at 16 13 49

Can we create a component similar to the component we use for account owner field or even better adapt it re-use that one? The main difference is that the search here would probably be done on the frontend side only.

A quick fix would be to just set a max-height + make this scrollable, but it would be best if we can add search.

Figma: https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?node-id=3007%3A69739&mode=dev

Screenshot 2024-01-10 at 16 21 24

### PR DESCRIPTION
**Pull request by ARSHIL1804**
fix #3356 and #3354 

- fix the Select Options Menu scrollable and added search
- add a select field in the filter and sort operation

### CODE REVIEW

**Comment by charlesBochet** [comment]
@ARSHIL1804 I'm taking a look at your PR tomorrow morning :)

**Comment by ARSHIL1804** [comment]
> @ARSHIL1804 I'm taking a look at your PR tomorrow morning :)

Sure, No issues.

File: packages/twenty-front/src/modules/object-record/select/types/SelectableOption.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { ThemeColor } from '@/ui/theme/constants/MainColorNames';
+
+export type SelectableOption = {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/select/types/SelectableOption.ts:3]: move this type to filter and can we infer it from another type?

**Review by charlesBochet — APPROVED** [review]
Follow-up tickets to create:
- add story on new components: MenuItemSelectTag + ObjectFilterDropdownSelect
- make sure that we can navigate with arrows in ObjectFilterDropdownSelect + SelectFieldInput

**Review by charlesBochet — APPROVED** [review]

**Comment by thomtrp** [comment]
@ARSHIL1804 we did a couple of changes and merged your work. Thank you for your contribution! 🔥

**Comment by ARSHIL1804** [comment]
Thank you for your appreciation. Currently I am busy with my own project. I will start making contributions again from this weekend🤝.

**Pull request merged**